src/test/java/org/broadinstitute/consent/http/service/dao/VoteServiceDAOTest.java

```diff
@@ -118,4 +131,106 @@ public void testUpdateVotesWithValue_closedElection() throws Exception {
+
+  @Test
+  public void testUpdateVotesWithValue_ClosedRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.CLOSED);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_CanceledRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.CANCELED);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_FinalRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.FINAL);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_PendingApprovalRPElection() throws Exception {
+    testUpdateVotesWithValue_RPElectionWithStatus(ElectionStatus.PENDING_APPROVAL);
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_MultipleRPElections() throws Exception {
+    User user = createUser();
+    DataAccessRequest dar = createDataAccessRequestV3();
+    DataSet dataset = createDataset();
+    Election election1 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election election2 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election election3 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    changeElectionStatus(election2, ElectionStatus.CANCELED);
+    changeElectionStatus(election3, ElectionStatus.CLOSED);
+    Vote vote1 = createDacVote(user.getDacUserId(), election1.getElectionId());
+    Vote vote2 = createDacVote(user.getDacUserId(), election2.getElectionId());
+    Vote vote3 = createDacVote(user.getDacUserId(), election3.getElectionId());
+    String rationale = "rationale";
+    initService();
+
+    List<Vote> votes = serviceDAO.updateVotesWithValue(List.of(vote1, vote2, vote3), true, rationale);
+    assertNotNull(votes);
+    assertFalse(votes.isEmpty());
+    votes.forEach(v -> assertTrue(v.getVote()));
+    votes.forEach(v -> assertEquals(rationale, v.getRationale()));
+  }
+
+  @Test
+  public void testUpdateVotesWithValue_MultipleElectionTypes() throws Exception {
+    User user = createUser();
+    DataAccessRequest dar = createDataAccessRequestV3();
+    DataSet dataset = createDataset();
+    Election rpElection1 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election rpElection2 = createRPElection(dar.getReferenceId(), dataset.getDataSetId());
+    Election accessElection = createAccessElection(dar.getReferenceId(), dataset.getDataSetId());
+    changeElectionStatus(rpElection1, ElectionStatus.CLOSED);
+    Vote vote1 = createDacVote(user.getDacUserId(), rpElection1.getElectionId());
+    Vote vote2 = createDacVote(user.getDacUserId(), rpElection2.getElectionId());
+    Vote vote3 = createDacVote(user.getDacUserId(), accessElection.getElectionId());
+    String rationale = "rationale";
+    initService();
+
+    List<Vote> votes = serviceDAO.updateVotesWithValue(List.of(vote1, vote2, vote3), true, rationale);
+    assertNotNull(votes);
+    assertFalse(votes.isEmpty());
+    votes.forEach(v -> assertTrue(v.getVote()));
```

Comment:
Forgot to mention this, instead of running two different loops across votes you can use bracket notation and group the assertions together in one loop.

So something like `votes.forEach(v -> { <assertions block> } )`